plat/kvm/x86: Provide access to ELF/PHDR structures

mogasergiu opened the pull request:
<!--

Thank you for opening a new PR to the Unikraft Open Source Project!  We welcome
new changes, features, fixes, and more!  Please fill in this form to indicate
the status of your PR.  Please ensure you have read the contribution guidelines
before opening a new PR as this will cover the PR process:

  https://unikraft.org/docs/contributing/

-->

### Prerequisite checklist

<!--
Please mark items appropriately:
-->

 - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
 - [x] Tested your changes against relevant architectures and platforms;
 - [x] Ran the [`checkpatch.uk`](https://github.com/unikraft/unikraft/blob/staging/support/scripts/checkpatch.uk) on your commit series before opening this PR;
 - [ ] Updated relevant documentation.


### Base target

 - Architecture(s): [e.g. `x86_64` or N/A]
 - Platform(s): [e.g. `kvm`, `xen` or N/A]
 - Application(s): [e.g. `app-python3` or N/A]


### Additional configuration

<!--
Please specify any additional configuration which is needed for this feature to
work or any new configuration parameters which are introduced by this PR.  This
will help during the review process.  For example:

 - `CONFIG_LIBUKDEBUG=y`

-->

### Description of changes

<!--
Please provide a detailed description of the changes made in this new PR.
-->

Programs and libc implementation can use these symbols to figure out the layout of the program in-memory. Usually, programs access this information using the `dl_iterate_phdr` function.

jschlumpp committed:
plat/kvm/x86: Provide access to ELF/PHDR structures

Programs and libc implementation can use these symbols to figure out the
layout of the program in-memory. Usually, programs access this
information using the `dl_iterate_phdr` function.

Signed-off-by: Marco Schlumpp <marco@unikraft.io>
Checkpatch-Ignore: FROM_SIGN_OFF_MISMATCH

michpappas reviewed (changes requested):
I think with the opportunity we should also do the same for `kvm/arm64` and `plat/xen` too, as any consuming code should work for all of these.

mogasergiu committed:
plat/kvm/arm: Provide access to ELF/PHDR structures

Programs and libc implementation can use these symbols to figure out the
layout of the program in-memory. Usually, programs access this
information using the `dl_iterate_phdr` function.

Signed-off-by: Sergiu Moga <sergiu@unikraft.io>

mogasergiu committed:
plat/xen: Provide access to ELF/PHDR structures

Programs and libc implementation can use these symbols to figure out the
layout of the program in-memory. Usually, programs access this
information using the `dl_iterate_phdr` function.

Signed-off-by: Sergiu Moga <sergiu@unikraft.io>

michpappas reviewed (approved):
Thanks!
Reviewed-by: Michalis Pappas <michalis@unikraft.io>
Approved-by: Michalis Pappas <michalis@unikraft.io>
